app/main/views/frameworks.py

```diff
@@ -212,32 +212,40 @@ def get_framework_suppliers(framework_slug):
         Framework.slug == framework_slug
     ).first_or_404()
 
-    # TODO: Work out a way to order by current_framework_agreement.signed_agreement_returned_at
-    # Listing agreements is something done for Admin only (suppliers only retrieve their individual agreements)
-    # CCS always want to work from the oldest returned date to newest, so order by ascending date
+    # we're going to need an alias for the current_framework_agreement join
+    cfa = orm.aliased(SupplierFramework._CurrentFrameworkAgreement)
+    supplier_frameworks = SupplierFramework.query.outerjoin(
+        cfa, SupplierFramework.current_framework_agreement
+    ).options(
+        orm.contains_eager(
+            SupplierFramework.current_framework_agreement, alias=cfa
+        )
+    )
 
-    # TODO: Work out a way to query.filter() on current_framework_agreement rather than manipulating with Python
-    supplier_frameworks = SupplierFramework.query.filter(
+    supplier_frameworks = supplier_frameworks.filter(
         SupplierFramework.framework_id == framework.id
+    ).order_by(
+        cfa.most_recent_signature_time.asc().nullsfirst(),
```

Comment:
Hmm, there's a subtle distinction here between ordering on the `most_recent_signature_time` (which could be time of signing or of countersigning), and the `signed_agreement_returned_at` time of the `cfa`.  Doing it by `most_recent_signature_time` as we have here will move a supplierFramework to the bottom of the list as soon as it's countersigned, whereas doing it by `signed_agreement_returned_at` will keep the same ordering as CCS get on with the countersigning.

I'm not sure which makes more sense.  Maybe it's fine as it is?  What do you think?